packages/lib/sdk/src/usql/query/Query.js

```diff
@@ -367,8 +367,12 @@ ${this.text.trim()}
 		);
 		return resolved;
 	};
-	fetch = async () => {
-		return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]).then(() => this.value);
+	fetch = () => {
+		const cols = this.#fetchColumns();
+		if (cols instanceof Promise) {
+			return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]).then(() => this.value);
```

Comment:
```suggestion
			return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]);
```
either return `this.value` in the sync version or stop returning `this.value` in either path